test/agentchat/test_groupchat.py

```diff
@@ -672,6 +672,38 @@ def test_clear_agents_history():
     ]
 
 
+def test_get_all_agents_in_manager():
+    def agent(name: str) -> autogen.ConversableAgent:
+        return autogen.ConversableAgent(
+            name=name,
+            max_consecutive_auto_reply=10,
+            human_input_mode="NEVER",
+            llm_config=False,
+        )
+
+    def team(name: str) -> autogen.ConversableAgent:
+        member1 = agent(f"member1_{name}")
+        member2 = agent(f"member2_{name}")
+
+        gc = autogen.GroupChat(agents=[member1, member2], messages=[])
+
+        return autogen.GroupChatManager(groupchat=gc, name=name, llm_config=False)
+
+    user = agent("user")
+    team1 = team("team1")
+    team2 = team("team2")
```

Comment:
Can we consider a new test case of nesting team2B under team2 and see that nested group chat works, please?

Reply:
I've added a test for chat functionality between agents within the main shared group chat, but managed by different group chat managers. Let me know what you think.